glusterblockrestd/middleware/req_logging.go

```diff
@@ -0,0 +1,28 @@
+package middleware
+
+import (
+	"net/http"
+
+	"github.com/gluster/gluster-block-restapi/pkg/context"
+
+	"github.com/gorilla/handlers"
+	"github.com/pborman/uuid"
+	log "github.com/sirupsen/logrus"
+)
+
+// WithReqLogger is a http middleware which will log incoming
+// http request.
+func WithReqLogger(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		var (
+			reqID     = uuid.NewRandom()
+			logWriter = log.WithField("reqid", reqID.String())
+			ctx       = r.Context()
+		)
+		{
```

Comment:
why this bracket is needed?

Reply:
block is not really doing anything here, I just gave it to separate the operation performed on the context object.

Reply:
it can be separated by newline also.